### PR TITLE
Add Template IDs to hellosign-sdk SignatureRequest

### PR DESCRIPTION
**Pull request by youngchen7**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [hellosign callback reference](https://developers.hellosign.com/api/reference/operation/apiAppCreateEventCallback/#!path=signature_request&t=request)
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/hellosign-sdk/hellosign-sdk-tests.ts [hunk 1]
```diff
@@ -113,6 +113,7 @@ const signatureRequestResponse: HelloSign.SignatureRequestResponse = {
         details_url: 'details_url',
         cc_email_addresses: ['cc_email_addresses'],
         signing_redirect_url: 'signing_redirect_url',
+	template_ids: ['template-id'],
```
Comment (DangerBotOSS) [inline types/hellosign-sdk/hellosign-sdk-tests.ts:116]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

File: types/hellosign-sdk/index.d.ts [hunk 2]
```diff
@@ -152,6 +152,7 @@ declare namespace HelloSign {
         details_url: string;
         cc_email_addresses: string | string[];
         signing_redirect_url: string;
+	template_ids: null | string[];
```
Comment (DangerBotOSS) [inline types/hellosign-sdk/index.d.ts:155]: - :no_entry_sign: The root .editorconfig style specifies spaces for whitespace. Please use spaces in new or changed types.

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## hellosign-sdk ([<kbd>unpkg</kbd>](https://unpkg.com/browse/hellosign-sdk@latest/))
was missing the following properties:
1. DEFAULT_HOST
1. DEFAULT_PORT
1. DEFAULT_BASE_PATH
1. DEFAULT_API_VERSION
1. DEFAULT_TIMEOUT

<details>
<summary>as well as these 5 other properties...</summary>
<p>PACKAGE_VERSION, USER_AGENT, USER_AGENT_SERIALIZED, HelloSignResource, resources</p>
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against f90697888f0691527fad4d071efa562583979b2c
</p>

File: types/hellosign-sdk/hellosign-sdk-tests.ts (hunk 1 above)
Comment (youngchen7) [reply types/hellosign-sdk/hellosign-sdk-tests.ts:116]: Fixed

File: types/hellosign-sdk/index.d.ts (hunk 2 above)
Comment (youngchen7) [reply types/hellosign-sdk/index.d.ts:155]: Fixed

**Comment by youngchen7** [comment]
The options for createEmbeddedWithTemplate and sendWithTemplate are also wrong - `cc_email_addresses` is an invalid parameter and it should be `ccs` (which is also typed incorrectly) 

https://developers.hellosign.com/api/reference/operation/signatureRequestCreateEmbeddedWithTemplate/#!ct=application/json&path=ccs&t=request

It should be `{role: string, email_address: string}[] | undefined` 
not `Record<string, { email_address: string }> | undefined`

**Pull request merged**